modules/gapi/test/infer/gapi_infer_ie_test.cpp

```diff
@@ -22,6 +22,10 @@
 #include "backends/ie/util.hpp"
 #include "backends/ie/giebackend/giewrapper.hpp"
 
+#ifdef HAVE_NGRAPH
+#include <ngraph/ngraph.hpp>
+#endif
```

Comment:
> @alalek These warnings come from `ngraph`
> 
> ```
>    572>C:\Program Files (x86)\IntelSWTools\openvino_2021.4.689\deployment_tools\ngraph\include\ngraph/op/abs.hpp(26,57): warning C4100: 'visitor': unreferenced formal parameter (compiling source file C:\build\precommit_custom_windows\opencv\modules\gapi\test\infer\gapi_infer_ie_test.cpp) [C:\build\precommit_custom_windows\build\modules\gapi\opencv_test_gapi.vcxproj]
> ```


Use this prolog code before ngraph include : https://github.com/opencv/opencv/blob/17234f82d025e3bbfbf611089637e5aa2038e7b8/modules/core/src/hal_replacement.hpp#L50-L59

and similar prolog code (check referenced file).

---

/cc @ilya-lavrenov FYI (public headers should not emit warnings)

Reply:
Added, works fine now, thanks!

Reply:
@alalek it's fixed in latest master (checked by https://github.com/openvinotoolkit/openvino/pull/8403 where I don't have to fix such issues)

Secondly, you can include IE headers as `SYSTEM`

Reply:
> as SYSTEM

Even on Windows and MSVC?

---

> in latest master

Upcoming release is 2021.4 LTS update. So we still forced to use/validate that.